### ISSUE TITLE
fixed learning rate vs multi step learning rate

### ISSUE THREAD

**Issue by twmht** [thughost2/Padam#2]
Hi,

as far as I know, the advantage of adam is that it can adjust the learning rate by itself, so it's called adaptive learning rate. So I can fixed the base learning rate to 0.01 in the whole training.

But in the code (https://github.com/uclaml/Padam/blob/master/run_cnn_test_cifar10.py#L128), I have no idea why you use multi step learning rate? Is it better than "fixed learning rate"?

**Comment by jinghuichen** [comment]
Yes. In training deep neural networks, people found that multi-step decaying learning is more helpful and give you better result. That's why modern CNN architecture papers love to use multi-step decaying learning rates. If you try with Adam, you will find the same effect. 

The adaptive learning rate means we adjust the actual learning rate according to previous gradient information, it is not contrary to the multi-step decaying mechanism. The theory, in general, will only require the learning rate to be none increasing.

**Comment by twmht** [comment]
ok

thank you

**Comment by ahundt** [comment]
@jinghuichen On this same topic, did you find cosine annealing or the 3 step lr did better with Padam?

**Comment by jinghuichen** [comment]
We didn't test with cosine annealing. Multi-step will give us better result compare with plain fixed learning rate.
